pkg/gardenlet/controller/vpaevictionrequirements/reconciler.go

```diff
@@ -0,0 +1,175 @@
+	err := r.SeedClient.Get(seedCtx, request.NamespacedName, vpa)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
+	}
+
+	log.Info("Reconciling vpa")
+
+	// double check just for fun: does the vpa have our label?
+	if !metav1.HasLabel(vpa.ObjectMeta, constants.LabelVPAEvictionRequirementDownscaleRestriction) {
+		err := fmt.Errorf("label %q not found, although marker label %q is present", constants.LabelVPAEvictionRequirementDownscaleRestriction, constants.LabelVPAEvictionRequirementsController)
+		log.Error(err, "Error while parsing the label value:")
+		return reconcile.Result{}, err
+	}
+	value := vpa.GetLabels()[constants.LabelVPAEvictionRequirementDownscaleRestriction]
+	log.Info("Found the label "+constants.LabelVPAEvictionRequirementDownscaleRestriction, "value", value)
+	switch value {
+	case constants.EvictionRequirementNever:
+		return r.reconcileVPAForDownscaleDisabled(ctx, vpa)
+	case constants.EvictionRequirementInMaintenanceWindowOnly:
+		return r.reconcileVPAForDownscaleInMaintenanceOnly(ctx, vpa)
+	default:
+		err := fmt.Errorf("unsupported label value found: %q, supported are only %q and %q", value, constants.EvictionRequirementNever, constants.EvictionRequirementInMaintenanceWindowOnly)
+		log.Error(err, "Error while parsing the label value:")
+		return reconcile.Result{}, err
+	}
+}
+
+func (r *Reconciler) reconcileVPAForDownscaleInMaintenanceOnly(ctx context.Context, vpa *vpaautoscalingv1.VerticalPodAutoscaler) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+	if !metav1.HasAnnotation(vpa.ObjectMeta, constants.AnnotationShootMaintenanceWindow) {
+		err := fmt.Errorf("didn't find maintenance window annotation, but VPA had label to be downscaled in maintenance only")
+		log.Error(err, "Error during reconciling for downscaling in maintenance only:")
+		return reconcile.Result{}, err
+	}
+
+	windowAnnotation := vpa.GetAnnotations()[constants.AnnotationShootMaintenanceWindow]
+	splitWindowAnnotation := strings.Split(windowAnnotation, ",")
+	if len(splitWindowAnnotation) != 2 {
+		err := fmt.Errorf("error during parsing the maintenance window from annotation. Value is not in format '<begin>,<end>': %q", windowAnnotation)
+		log.Error(err, "Error during reconciling for downscaling in maintenance only:")
+		return reconcile.Result{}, err
+	}
+	maintenanceTimeWindow, err := timewindow.ParseMaintenanceTimeWindow(splitWindowAnnotation[0], splitWindowAnnotation[1])
+	if err != nil {
+		log.Error(err, "Error during parsing the maintenance window from start and end time", "begin", splitWindowAnnotation[0], "end", splitWindowAnnotation[1])
+		return reconcile.Result{}, err
+	}
+
+	if isNowInMaintenanceTimeWindow := maintenanceTimeWindow.Contains(r.Clock.Now()); isNowInMaintenanceTimeWindow {
+		log.Info("Shoot is inside maintenance window, removing the EvictionRequirement to allow downscaling", "shoot-namespace", vpa.GetNamespace(), "maintenanceWindow", maintenanceTimeWindow)
```

Comment:
```suggestion
		log.Info("Shoot is inside maintenance window, removing the EvictionRequirement to allow downscaling", "maintenanceWindow", maintenanceTimeWindow)
```

This information is already present on the values